common/src/main/java/com/wynntils/mc/mixin/ChestMenuMixin.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.mc.mixin;
+
+import com.wynntils.mc.EventFactory;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.inventory.AbstractContainerMenu;
+import net.minecraft.world.inventory.ChestMenu;
+import net.minecraft.world.inventory.MenuType;
+import net.minecraft.world.item.ItemStack;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+@Mixin(ChestMenu.class)
+public abstract class ChestMenuMixin extends AbstractContainerMenu {
+    protected ChestMenuMixin(MenuType<?> menuType, int i) {
+        super(menuType, i);
+    }
+
+    @Inject(method = "quickMoveStack", at = @At("HEAD"))
+    public void onQuickMoveStack(Player player, int index, CallbackInfoReturnable<ItemStack> cir) {
+        EventFactory.onChestMenuQuickMove(this.containerId);
```

Comment:
I think you can do a double cast like this:
```
        EventFactory.onChestMenuQuickMove((ChestMenu)(Object)this.containerId);
```
and then you don't need to extend the abstract class, and can remove the constructor.

Reply:
Double casts can be done